llmstack/datasources/handlers/databases/sql.py

```diff
@@ -72,6 +72,9 @@ class SQLDatabaseSchema(DataSourceSchema):
         widget="connection",
         advanced_parameter=False,
         description="Use your authenticated connection to the database",
+        filters={
+            "base_connection_type": ConnectionType.CREDENTIALS,
```

Comment:
Let's do filters something like `base_connection_type/provider_slug/connection_type_slug` and make them a list. And we may skip providing `provider_slug` or `connection_type_slug` and we filter accordingly.

For example, if I give `filters=["credentials"]`, it should filter by `base_connection_type=credentials`. if I do `filters=["credentials/promptly", "credentials/apollo"]`, it should provide all credential type connections from both Promptly and Apollo.